app/src/main/java/com/example/weatherapp/model/City.kt

```diff
@@ -0,0 +1,24 @@
+package com.example.weatherapp.model
+
+data class City(
+    var admin1 code: String?, // AL
+    var admin2 code: Int?, // 117
+    var admin3 code: Int?, // 21504
+    var admin4 code: String?,
+    var alternatenames: String?, // Buxahatchie,Calera,Kalera,Lime Kiln,Lime Plant,Lime Station,ka lai la,kalra,kalyra,kyalera,Калера,كاليرا,کالرا,क्यालेरा,卡莱拉
+    var asciiname: String?, // Calera
+    var cc2: String?,
+    var country code: String?, // US
+    var dem: Int?, // 156
+    var elevation: Int?, // 154
+    var feature class: String?, // P
+    var feature code: String?, // PPL
+    var geonameid: Int?, // 4053050
+    var imageURLs: ImageURLs?,
+    var latitude: Double?, // 33.1029
+    var longitude: Double?, // -86.7536
+    var modification date: String?, // 3/9/17
+    var name: String?, // Calera
+    var population: Int?, // 13213
+    var timezone: String? // America/Chicago
+)
```

Comment:
Should the variable names be renamed without spaces such as admin1 code to admin1_code or admin1Code? Same for the admin2 code, admin3 code, admin4 code, country code, feature class, feature code, and modification date?

Reply:
That will be the case @mysia-rox0515 if this will be a "presentation layer" or an information that is produce by us (the code or data flow itself). This current model/City.kt will be used for the parsing after retrofit getting data from the server, so, this fields needs to match exactly as the definition in the backend controller.

@arabking20 
This could be a proposal, you can try and do some changes using the annotation
SerializedName.

Reply:
At this point I don't know which data will need it. I just created a variable for each KEY of the json file. who ever works in the  front end will chose which data want to display. we don't need  all the variables I  created in the Model packages. And @mysia-rox0515 Questions It is easy to fix it. We can use A custom name for our KEY. we can change KEY name using a SerializedName Annotation. will fix easily.